numpy/core/src/multiarray/conversion_utils.c

```diff
@@ -497,21 +499,35 @@ PyArray_SelectkindConverter(PyObject *obj, NPY_SELECTKIND *selectkind)
 static int searchside_parser(char const *str, Py_ssize_t length, void *data)
 {
     NPY_SEARCHSIDE *side = (NPY_SEARCHSIDE *)data;
+    int is_exact = 0;
 
     if (length < 1) {
         return -1;
     }
     else if (str[0] == 'l' || str[0] == 'L') {
         *side = NPY_SEARCHLEFT;
-        return 0;
+        is_exact = (length == 4 && strcmp(str, "left") == 0);
     }
     else if (str[0] == 'r' || str[0] == 'R') {
         *side = NPY_SEARCHRIGHT;
-        return 0;
+        is_exact = (length == 5 && strcmp(str, "right") == 0);
     }
     else {
         return -1;
     }
+
+    /* Filters out the case sensitive/non-exact
+     * match inputs and other inputs and outputs DeprecationWarning */
+    if (!is_exact) {
+        /* NumPy 1.20, 2020-05-19 */
+        if (DEPRECATE("inexact matches and case insensitive matches "
+                      "for search side are deprecated. Please use "
+                      "one of 'left' or 'right' instead") < 0) {
```

Comment:
```suggestion
                      "one of 'left' or 'right' instead.") < 0) {
```

Reply:
And below. Either than, or rephrase this to use `deprecate, please` instead of `deprecated. Please`.